src/ogdc_runner/recipe/simple.py

```diff
@@ -1,269 +1,170 @@
-        err_msg = "Failed to check if data have been published"
-        raise OgdcWorkflowExecutionError(err_msg)
-
-    assert result in ("yes", "no")
-
-    # Cleanup this workflow, it is no longer needed
-    ARGO_WORKFLOW_SERVICE.delete_workflow(workflow_name)
-
-    return result == "yes"
-
-
-def data_already_published(*, recipe_config: RecipeConfig, overwrite: bool) -> bool:
-    """Check for the existence of published data for the given
-    recipe and optionally remove it.
-
-    If `overwrite=True`, this function will remove any existing published data
-    for the provided recipe.
-
-    Returns `True` if data have already been published for the given recipe,
-    otherwise `False`.
-    """
-    if overwrite:
-        # If `overwrite` is True, remove the existing data and return `False`.
-        _remove_existing_published_data(
-            recipe_config=recipe_config,
-        )
-        return False
-
-    return _check_for_existing_published_data(
-        recipe_config=recipe_config,
-    )
+from ogdc_runner.recipe.viz_workflow import submit_viz_workflow_recipe
 
 
 def make_and_submit_simple_workflow(
     recipe_config: RecipeConfig,
     wait: bool,
+    custom_image: str | None = None,
+    custom_tag: str | None = None,
+    custom_namespace: str | None = None,
+    update_global: bool = False,
 ) -> str:
     """Create and submit an argo workflow based on a simple recipe.
 
+    Args:
+        recipe_config: The recipe configuration
+        wait: Whether to wait for the workflow to complete
+        custom_image: Optional custom image to use for all containers
+        custom_tag: Optional custom tag for the image
+        custom_namespace: Optional custom namespace for the workflow
+        update_global: If True, update the global image config; if False, only apply to this workflow
+
     Returns the name of the workflow as a str.
     """
-    commands = _cmds_from_simple_recipe(recipe_config.recipe_directory)
+    # Parse commands from the simple recipe file
+    commands = parse_commands_from_recipe_file(
+        recipe_config.recipe_directory,
+        SIMPLE_RECIPE_FILENAME,
+    )
 
     with Workflow(
         generate_name=f"{recipe_config.id}-",
         entrypoint="steps",
         workflows_service=ARGO_WORKFLOW_SERVICE,
     ) as w:
+        # Apply custom configuration if provided
+        apply_custom_container_config(
+            custom_image=custom_image,
+            custom_tag=custom_tag,
+            custom_namespace=custom_namespace,
+            update_global=update_global,
+        )
+
+        # Create command templates
         cmd_templates = []
         for idx, command in enumerate(commands):
-            cmd_template = _make_cmd_template(name=f"run-cmd-{idx}", command=command)
+            cmd_template = make_cmd_template(
+                name=f"run-cmd-{idx}",
+                command=command,
+                custom_image=custom_image,
+                custom_tag=custom_tag,
```

Comment:
Why do `custom_image` and `custom_tag` need to be passed into this function when we apply the configs in `apply_custom_container_config` above? Can we avoid/remove this redundancy?

Reply:
Same with other template functions below. 

If this really is data that needs to be maintained and passed along in multiple locations, it would be good to consolidate and, as mentioned above, maybe make these part of the recipe definition.